libmcount/plthook.c

```diff
@@ -744,6 +744,16 @@ unsigned long plthook_entry(unsigned long *ret_addr, unsigned long child_idx,
 
 	recursion = false;
 
+#if defined(__i386__)
+	if (child_idx > 0) {
+		if (child_idx % 8) {
+			pr_err_ns("the malformed child idx : %lx\n", child_idx);
+		}
+		child_idx = child_idx / 8;
+	}
+#endif
```

Comment:
this needs some explanation - why it's invalid?

Also I'd like to avoid such ifdef-ery in the .c file as long as possible.  You may want to define a mcount_arch_XXX function for it and make them NOP for other archs.

Reply:
I do not know why, but the value is set to a multiple of 8.